ios/Trifle/Sources/Trifle.swift

```diff
@@ -62,24 +62,54 @@ public class Trifle {
     }
     
     /**
-     // TODO(dcashman): define message format //
      Sign the provided data with the provided key, including appropriate Trifle
      metadata, such as the accompanying certificate.
 
      - parameters: data - raw data to be signed.
      - parameters: keyHandle - key handle used for the signing.
-     - parameters: certificate - certificate to be included in the SignedData message.
+     - parameters: certificate list - certificate chain to be included in the SignedData message.
         Must match the key in keyHandle.
 
      - returns:`SignedData` - signed data message in the Trifle format.
     */
     public func createSignedData(
         data: Data,
         keyHandle: KeyHandle,
-        certificate: Certificate
-    ) -> SignedData {
-        // TODO: IMPLEMENT
-        SignedData()
+        certificate: Array<Certificate>
+    ) throws -> SignedData {
+
+        guard !data.isEmpty && !certificate.isEmpty else {
```

Comment:
instead of just checking for an empty list, we can
```
        guard let leafCert = certificate.first, !data.isEmpty else {
            throw TrifleError.invalidInput("Data or Certifcate should not be empty.")
        }
```